src/aiida/storage/sqlite_dos/backend.py

```diff
@@ -225,3 +234,36 @@ def _get_mapper_from_entity(entity_type: 'EntityTypes', with_pk: bool):
         mapper = inspect(model).mapper  # type: ignore[union-attr]
         keys = {key for key, col in mapper.c.items() if with_pk or col not in mapper.primary_key}
         return mapper, keys
+
+    def _backup(
+        self,
+        dest: str,
+        keep: Optional[int] = None,
+    ):
+        try:
+            backup_manager = backup_utils.BackupManager(dest, keep=keep)
+            backup_manager.backup_auto_folders(lambda path, prev: self._backup_storage(backup_manager, path, prev))
```

Comment:
This line is quite hard to parse. But I am not sure sure how to make it better, it seems the complexity stems from `backup_auto_folders` API.

Reply:
It does. This code is essentially borrowed from the `disk_objectstore` to automatically maintain the folder structure of multiple backup copies where the latest is symlinked and where rsync can make use of incremental backups.